geomesa-core/src/main/scala/org/locationtech/geomesa/core/iterators/GeomesaFilteringIterator.scala

```diff
@@ -82,6 +75,12 @@ abstract class GeomesaFilteringIterator
    */
   def setTopConditionally(): Unit
 
-  override def deepCopy(env: IteratorEnvironment) =
+  override def deepCopy(env: IteratorEnvironment): SortedKeyValueIterator[Key, Value] =
     throw new UnsupportedOperationException("GeoMesa iterators do not support deepCopy")
 }
+
+trait HasSourceIterator {
```

Comment:
Ok, I see what's going on here . . . the actual iterator is composed and shares these vars which get set in various places.  I find this a bit confusing - there's a lot of places where a mutation occurs - but off the top of my head I can't think of a better way.